onmt/opts.py

```diff
@@ -643,30 +644,42 @@ def _add_decoding_opts(parser):
                    "target token. If it is not provided (or the identified "
                    "source token does not exist in the table), then it "
                    "will copy the source token.")
+    group.add('--ban_unk_token', '-ban_unk_token',
+              action="store_true",
+              help="Prevent unk token use by setting unk proba to 0")
     group.add('--phrase_table', '-phrase_table', type=str, default="",
               help="If phrase_table is provided (with replace_unk), it will "
                    "look up the identified source token and give the "
                    "corresponding target token. If it is not provided "
                    "(or the identified source token does not exist in "
                    "the table), then it will copy the source token.")
+    group.add('--beam_size', '-beam_size', type=int, default=5,
```

Comment:
Not sure why this was moved here.

Reply:
I used the beam_size option in greedy search because we can generate several examples in one pass with sampling too. Hence this argument is not beam search specific anymore.
I can't duplicate the option.
I can create another option param specific to sampling but it will be used exactly like beam_size and will be named beam_size in shared part of the code as well like DecodeStrategy

Reply:
Ok, makes sense. Maybe just put a short comment here for those who might be asking similar questions.

Reply:
I found a way of showing that the parameter is shared in the code and in the -h, let me now what you think about it https://github.com/OpenNMT/OpenNMT-py/pull/1958/commits/94cfe91039581a6b67bde79546b768e517caaa0c#diff-c7951b1c7894cf513192baf6a05cebd25719e465c9edef24b102570877aec3a1R686